lib/elixir/lib/module/locals_tracker.ex

```diff
@@ -105,28 +105,31 @@ defmodule Module.LocalsTracker do
     undefined =
       for {pair, _, meta, _} <- all_defined,
           {local, line, macro_dispatch?} <- out_neighbours(bag, {:local, pair}),
-          error = undefined_local_error(set, local, macro_dispatch?),
+          error = undefined_local_error({set, bag}, local, macro_dispatch?),
           do: {build_meta(line, meta), local, error}
 
     :lists.usort(undefined)
   end
 
-  defp undefined_local_error(set, local, true) do
+  defp undefined_local_error({set, _bag}, local, true) do
     case :ets.member(set, {:def, local}) do
       true -> false
       false -> :undefined_function
     end
   end
 
-  defp undefined_local_error(set, local, false) do
-    try do
-      if :ets.lookup_element(set, {:def, local}, 2) in @defmacros do
-        :incorrect_dispatch
-      else
-        false
-      end
-    catch
-      _, _ -> :undefined_function
+  defp undefined_local_error({set, bag}, local, false) do
+    case :ets.lookup(set, {:def, local}) do
+      [] -> :undefined_function
+      [{_, kind, _, _, _, _}] when kind in @defmacros -> :incorrect_dispatch
+      _other -> bodiless_local_error(bag, local)
```

Comment:
I added the check here to be cheaper, but I don't know if it makes sense to put all together and still keep the same name for `undefined_local_error`

Reply:
Let's rollback to `ets:lookup_element` because it loads less data from the table, even if the code is ultimately uglier.